File: lib/ts/src/internal/binary/PackBody.ts
//|
//|  Copyright The Telepact Authors
//|  SPDX-License-Identifier: Apache-2.0
//|

import { pack } from './Pack.js';

export function packBody(body: Map<any, any>): Map<any, any> {
    const result: Map<any, any> = new Map();

    for (const [key, value] of body.entries()) {
        const packedValue = pack(value);
        result.set(key, packedValue);
    }

    return result;
}

File: lib/go/internal/binary/PackBody.go
//|
//|  Copyright The Telepact Authors
//|  SPDX-License-Identifier: Apache-2.0
//|

package binary

// PackBody packs the message body map into the compact binary representation when possible.
func PackBody(body map[any]any) (map[any]any, error) {
	result := make(map[any]any, len(body))
	for key, value := range body {
		packedValue, err := Pack(value)
		if err != nil {
			return nil, err
		}
		result[key] = packedValue
	}
	return result, nil
}
